test/dutchExchange.js

```diff
@@ -53,8 +53,17 @@ const setAndCheckAuctionStarted = async (ST, BT) => {
   const startingTimeOfAuction = (await dx.auctionStarts.call(ST.address, BT.address)).toNumber()
 
   // wait for the right time to send buyOrder
-  await wait(startingTimeOfAuction - timestamp() + 6 * 3600)
-  assert.equal(timestamp() > startingTimeOfAuction, true)
+  await wait(startingTimeOfAuction - timestamp() )
+  assert.equal(timestamp() >= startingTimeOfAuction, true)
+}
+
+const waitUntilPriceIsXPercentOfPreviousPrice = async (ST, BT, p, auctionIndex) => {
+  const startingTimeOfAuction = (await dx.auctionStarts.call(ST.address, BT.address)).toNumber()
+  const closingPrices = (await dx.closingPrices.call(ST.address, BT.address, auctionIndex-1))
+  const timeToWaitFor = (86400-p*43200)/(1+p)+startingTimeOfAuction
```

Comment:
@josojo although this is super internal, can you give a quick rundown of what exactly is happening here with `p`? What is `p`?